consensus/istanbul/backend/handler.go

```diff
@@ -221,9 +219,6 @@ func (sb *Backend) NewWork() error {
 // the parent block, because different nodes circulate different versions.
 // The bitmap of signed validators only becomes canonical when the child block is proposed.
 func (sb *Backend) UpdateMetricsForParentOfBlock(child *types.Block) {
-	sb.coreMu.RLock()
-	defer sb.coreMu.RUnlock()
-
```

Comment:
The mutex here was synchronising over a large block of code, are you sure that it wasn't necessary?

Reply:
coreMu is a bit confusing naming, it actually handles the `coreStarted`, you can check how it's used in other occurrances.
This Lock/Unlock you are referring to was introduced in [#515](https://github.com/celo-org/celo-blockchain/commit/046fabe1317c45882bd67ea518b0ab7066bdc30f#diff-2e3562d02008c09a2d82cafb85029a9331bd9d87679e66db063da5c08acd6436R197-R198), along with coreStarted.
And then, in this [#942](https://github.com/celo-org/celo-blockchain/commit/ae4886abfaa3fbfcddcd4e2873598798d340450d#diff-2e3562d02008c09a2d82cafb85029a9331bd9d87679e66db063da5c08acd6436R315-R322), `coreStarted` was removed, but lock/unlock was still here, until now.
Hence I think `coreStarted` is actually irrelevant to its following executions, so I removed it.

Reply:
Hi @YoduYodu I think you are referring to the other instance of lock/lock in `NewChainHead`

I think this use of coreMu was actually introduced in:
* #1029

Although the lock and unlock were not actually added from the perspective of git, the function around the lock and unlock changed completely, so conceptually that is where this instance of lock and unlock got added. And there when that change was made there was no reference to `coreStarted` inside `UpdateMetricsForParentOfBlock`. So I would suggest not removing this instance unless you can show that it is not actually needed.